src/flags.rs

```diff
@@ -21,29 +26,44 @@ pub struct DenoFlags {
   pub version: bool,
   pub reload: bool,
   pub recompile: bool,
-  pub allow_write: bool,
-  pub allow_net: bool,
-  pub allow_env: bool,
+  allow_write: bool,
+  allow_net: bool,
+  allow_env: bool,
   pub deps_flag: bool,
   pub types_flag: bool,
 }
 
-pub fn process(flags: &DenoFlags) {
-  if flags.help {
-    print_usage();
-    exit(0);
+impl DenoFlags {
+
+  pub fn permissions_check_write(&self, filename: &str) -> DenoResult<()> {
+    if self.allow_write { return Ok(()) };
+    // TODO get location (where access occurred)
+    permission_prompt(format!("Deno requests write access to \"{}\".", filename))
   }
```

Comment:
I think `permissions_check_write` and friends should not be methods of the Flags object, but rather standalone functions. Probably best to put them in `src/errors.rs` or even create a new `src/permissions.rs`

Reply:
The benefit of them being here is the `allow_write` field and friends are private, so permissions must be checked via `permissions_check_write`.

Reply:
we should probably have a standalone permissions struct - rather than using flags. But that should be done separately

Reply:
move to a new struct. Though in future this shouldn't be tacked on the DenoFlags, but be independent/new argument to Isolate/IsolateState. As a mutex.